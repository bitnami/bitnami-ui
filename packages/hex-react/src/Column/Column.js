import React from 'react';
import PropTypes from 'prop-types';
import cs from 'classnames';

const Column = props => {
  const { align, span, className, type, children, ...attributes } = props;

  const css = cs(className, `col-${span}`, {
    [`align-${align}`]: align != null,
  });

  return React.createElement(
    type,
    {
      className: css,
      ...attributes,
    },
    children,
  );
};

Column.propTypes = {
  /**
   * Set the HTML tag that will be rendered
   */
  type: PropTypes.oneOf(['div', 'section', 'article', 'main']),
  /**
   * Define the space the column takes in the parent row. By default, 12 is the max.
   */
  span: PropTypes.number.isRequired,
  /**
   * Vertical alignment for the column in the row.
   */
  align: PropTypes.oneOf(['start', 'center', 'end']),
  /**
   * Other CSS classes for the component
   */
  className: PropTypes.string,
  /**
   * Any element that will be wrapped by the Box
   */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
};

Column.defaultProps = {
  type: 'div',
  align: undefined,
  className: '',
  children: undefined,
};

export default Column;
