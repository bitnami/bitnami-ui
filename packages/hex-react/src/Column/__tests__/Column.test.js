import React from 'react';
import { shallow } from 'enzyme';

import Column from '../Column';

describe('Column', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Column span={12}>Test</Column>);
    expect(wrapper).toMatchSnapshot();
  });

  it('shows the children', () => {
    const text = 'Test';
    const wrapper = shallow(<Column span={6}>{text}</Column>);
    expect(wrapper.text()).toEqual(text);
  });

  it('sets the correct span number', () => {
    const wrapper = shallow(<Column span={6}>Test</Column>);
    expect(wrapper.hasClass('col-6')).toBe(true);
  });

  it('adds a custom CSS class', () => {
    const css = 'test';
    const wrapper = shallow(
      <Column span={12} className={css}>
        Test
      </Column>,
    );
    expect(wrapper.hasClass(css)).toBe(true);
    expect(wrapper.hasClass('col-12')).toBe(true);
  });

  it('sets CSS class for alignment', () => {
    const wrapper = shallow(
      <Column span={12} align="center">
        Test
      </Column>,
    );
    expect(wrapper.hasClass('align-center')).toBe(true);
  });
});
