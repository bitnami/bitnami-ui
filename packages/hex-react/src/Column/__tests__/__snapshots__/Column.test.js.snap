// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column renders correctly 1`] = `
<div
  className="col-12"
>
  Test
</div>
`;
